homeassistant/components/media_player/cambridgeaudio.py

```diff
@@ -0,0 +1,382 @@
+
+# supported features when device is configured as pre-amp
+SUPPORT_CAMBRIDGE_PREAMP = SUPPORT_VOLUME_MUTE | SUPPORT_VOLUME_SET |\
+                           SUPPORT_VOLUME_STEP
+
+# supported features for file-based media (i.e. not internet streams)
+SUPPORT_CAMBRIDGE_FILEMEDIA = SUPPORT_SEEK | SUPPORT_PAUSE |\
+                              SUPPORT_NEXT_TRACK | SUPPORT_PREVIOUS_TRACK |\
+                              SUPPORT_SHUFFLE_SET
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Cambridge Audio platform."""
+    from stream_magic import device as cadevice
+    from stream_magic import discovery as ca
+
+    host = config.get(CONF_HOST)
+    name = config.get(CONF_NAME)
+
+    poweroff_command = config.get(CONF_COMMAND_OFF, 'OFF')
+
+    hosts = []
+    sm = ca.StreamMagic()   # pylint: disable=C0103
+
+    # found a new host that was specified in config file
+    if CONF_HOST in config and host not in KNOWN_HOSTS:
+        try:
+            # discover() always returns a list, even for a single host
+            devices = sm.discover(host=host)
+            if devices:
+                dev = devices[0]
+                addr, port = dev[0][0:2]
+                desc = dev[1]['server']
+                scpd_url = dev[1]['location']
+                if not name:
+                    name = desc
+
+                smdevice = cadevice.StreamMagicDevice(addr, port,
+                                                      desc, scpd_url)
+                hosts.append(CADevice(smdevice,
+                                      config.get(CONF_SOURCES),
+                                      poweroff_command,
+                                      name=name))
+                KNOWN_HOSTS.append(host)
+                _LOGGER.debug("Added StreamMagic device with ip %s (%s)",
+                              addr, desc)
+        except OSError:
+            _LOGGER.debug("Unable to connect to device at %s", host)
+
+    elif discovery_info is not None:    # netdisco found the device
+        addr = discovery_info.get('host')
+        port = discovery_info.get('port')
+
+        # if a name was set in the config, use it
+        if name is None:
+            name = discovery_info.get('name')
+
+        scpd_url = discovery_info.get('ssdp_description')
+        if addr not in KNOWN_HOSTS:
+            smdevice = cadevice.StreamMagicDevice(addr, port, name, scpd_url)
+            hosts.append(CADevice(smdevice, config.get(CONF_SOURCES),
+                                  poweroff_command, name=name))
+            KNOWN_HOSTS.append(host)
+            _LOGGER.debug("Added StreamMagic device with ip %s (%s)",
+                          addr, name)
+    add_devices(hosts, True)
+
+
+class CADevice(MediaPlayerDevice):
+    """Representation of a Cambridge Audio Network Audio Player device."""
+
+    def __init__(self, smdevice, sources, poweroff_command, name=None):
```

Comment:
`sources` is never used.

Seems like `name` can be made a regular positional argument, instead of a keyword argument. It's supplied in all cases.